### PR TITLE
Template load

### PR DESCRIPTION
**Pull request by iocanel**
An different approach to #664

This pretty much adds the ability to load() non-Template resource as Template using:

     client.templates().load(is)

### CODE REVIEW

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/LoadAsTemplateTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    final AtomicBoolean userIsRoot = new AtomicBoolean(false);
+    final AtomicBoolean passwordIsNotNull = new AtomicBoolean(false);
+    new KubernetesListBuilder(list).accept(new TypedVisitor<EnvVarBuilder>() {
+
+      @Override
+      public void visit(EnvVarBuilder element) {
+        if (element.getName().equals("USERNAME")) {
+          userIsRoot.set(element.getValue().equals("root"));
+        } else if (element.getName().equals("PASSWORD")) {
+          passwordIsNotNull.set(element.getValue() != null);
```
Comment (jimmidyson) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/LoadAsTemplateTest.java]: Check length too perhaps to make sure it's got a generated value, not just empty string?

File: openshift-client/src/main/java/io/fabric8/openshift/client/dsl/Processable.java [hunk 2]
```diff
@@ -13,13 +13,31 @@
...
+
+  T process(File f);
+
+  T process(InputStream is);
+
+  T process(Map<String, String> map);
+
   T process(ParameterValue... values);
 
+  T processLocally(File f);
```
Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/dsl/Processable.java]: Can you add comments describing difference between `process` and `processLocally` please?

**Comment by iocanel** [comment]
@jimmidyson: Applied your feedback. Want to have a second look?

**Comment by jimmidyson** [comment]
LGTM thanks! [merge]

**Pull request merged**